### PR TITLE
Option to make tables separate node types

### PR DESCRIPTION
**Pull request by kevee**
Per conversation in #51, this PR:

- Adds a new `createSeparateNodeType` boolean option to tables
- When in use, it appends the table query name, or if unavailable, the table name to the `Airtable` node type (so a table `Fruit` would be under `allAirtableFruit`)

### CODE REVIEW

**Comment by kevee** [comment]
Good catch, added.

**Comment by jbolda** [comment]
Hmm, well Travis is failing and it does appear to be related, but I haven't figured what the issue is. Could you add the small example checking both `Airtable` and `AirtableField` change correctly. I can then copy it into my test base and we can add it to the travis.yml to have it run against the CI on future PRs so we don't accidentally break it.

File: README.md [hunk 1]
```diff
@@ -34,6 +34,7 @@ plugins: [
           tableName: `YOUR_TABLE_NAME`,
           tableView: `YOUR_TABLE_VIEW_NAME`, // optional
           queryName: `OPTIONAL_NAME_TO_IDENTIFY_TABLE`, // optional
+          createSeparateNodeType: true, // optional - makes all records in this table a separate node type, based on your tableView, or if not present, tableName, e.g. a table called "Fruit" would become "allAirtableFruit"
```
Comment (jbolda) [inline README.md]: Might be worth mentioning that the default is false. Also, we should probably add another section in the docs on _why_ you would care to do this. We could even point back to the issues regarding it.

File: gatsby-node.js [hunk 2]
```diff
@@ -146,6 +148,10 @@ exports.sourceNodes = async (
       store,
       cache
     });
+    
+    const nodeType = (row.separateNodeType) ?
```
Comment (jbolda) [inline gatsby-node.js]: The only issue I can think of is that that isn't falsey when we are expecting it to be (in regards to the failing test). Could it be an issue with an unresolved promise?

**Comment by jbolda** [comment]
Oh, and let's do a minor bump on this. Feel free to include that in this PR.

**Comment by kevee** [comment]
Will do! We just did a big launch so I'll probably not get to it until next week.

**Comment by jbolda** [comment]
Just checking in to make sure you are not waiting on a review.

**Comment by jbolda** [comment]
Checking in on this again. Still busy with that launch?

**Comment by ndrabins** [comment]
Any chance this is still possible to add in? Would love to be able to separate tables by node name. 

I'd be willing to take this on if no one else is intereseted. Thanks

**Comment by jbolda** [comment]
I would love to see this get in @ndrabins. We can merge this PR into a sperate feature branch and you can work off that. Otherwise you can start fresh if that seems to best course of action.

**Comment by ndrabins** [comment]
Awesome, sounds good @jbolda. I should be able to get to this either this weekend or next.

**Comment by ndrabins** [comment]
@jbolda hey! So I tried working on this PR this weekend starting from here. Immediately hit a roadblock in that even when I `npm link` I get an error about not being able to find the plugin. Any ideas?

Also I haven't tested it yet, but this PR seems pretty solid to my beginner eyes. Was there something specific the original author was missing or just a broken test?

**Comment by jbolda** [comment]
@ndrabins I think it is pretty close. We needed to get the existing tests to pass and add a new test to check this implementation.

Hmm, that is pretty odd. We are working through an issue on the newest gatsby via #117 that may be related?

Since it sounds like you are in agreement that what is here is good so far, I will go ahead and merge this into a separate feature branch sometime today that we can work on. Sound good to you?

**Comment by ndrabins** [comment]
Yea that sounds good. I'll check that out then and go from there. Will try again to see if I can get the linking to work. Thanks!

**Pull request merged**